geode-assembly/src/acceptanceTest/java/org/apache/geode/metrics/function/executions/FunctionExecutionsTimerLonerTest.java

```diff
@@ -0,0 +1,364 @@
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import org.apache.geode.cache.client.ClientCache;
+import org.apache.geode.cache.client.ClientCacheFactory;
+import org.apache.geode.cache.client.Pool;
+import org.apache.geode.cache.client.PoolManager;
+import org.apache.geode.cache.execute.Execution;
+import org.apache.geode.cache.execute.Function;
+import org.apache.geode.cache.execute.FunctionService;
+import org.apache.geode.internal.AvailablePortHelper;
+import org.apache.geode.metrics.MetricsPublishingService;
+import org.apache.geode.metrics.SimpleMetricsPublishingService;
+import org.apache.geode.rules.ServiceJarRule;
+import org.apache.geode.test.junit.rules.gfsh.GfshRule;
+
+/**
+ * Acceptance tests for function executions timer on a loner server with no locator
+ */
+public class FunctionExecutionsTimerLonerTest {
+
+  private int serverPort;
+  private int jmxRmiPort;
+  private Path serverFolder;
+  private Path serviceJarPath;
+  private Path functionHelpersJarPath;
+  private ClientCache clientCache;
+  private Pool serverPool;
+  private String connectCommand;
+  private String startServerCommandWithStatsEnabled;
+  private String startServerCommandWithStatsDisabled;
+  private String startServerCommandWithTimeStatsDisabled;
+  private String stopServerCommand;
+
+  @Rule
+  public GfshRule gfshRule = new GfshRule();
+
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Rule
+  public ServiceJarRule serviceJarRule = new ServiceJarRule();
+
+  @Before
+  public void setUp() throws IOException {
+    int[] ports = AvailablePortHelper.getRandomAvailableTCPPorts(2);
+
+    serverPort = ports[0];
+    jmxRmiPort = ports[1];
+
+    serverFolder = temporaryFolder.newFolder("server").toPath().toAbsolutePath();
+
+    serviceJarPath = serviceJarRule.createJarFor("services.jar",
+        MetricsPublishingService.class, SimpleMetricsPublishingService.class);
+
+    functionHelpersJarPath =
+        temporaryFolder.getRoot().toPath().resolve("function-helpers.jar").toAbsolutePath();
+    writeJarFromClasses(functionHelpersJarPath.toFile(), FunctionToTimeWithResult.class,
+        GetFunctionExecutionTimerValues.class, ExecutionsTimerValues.class);
+
+    startServerCommandWithStatsEnabled = startServerCommand(false, true);
+    startServerCommandWithStatsDisabled = startServerCommand(true, true);
+    startServerCommandWithTimeStatsDisabled = startServerCommand(false, false);
+
+    stopServerCommand = "stop server --dir=" + serverFolder;
+    connectCommand = "connect --jmx-manager=localhost[" + jmxRmiPort + "]";
+  }
+
+  @After
+  public void tearDown() {
+    stopServer();
+  }
+
+  @Test
+  public void timersNotRegisteredIfFunctionDeployedButNotExecuted() {
+    startServerWithStatsEnabled();
+
+    deployFunction(FunctionToTimeWithResult.class);
+
+    Assertions.assertThat(getExecutionsTimerValuesFor(FunctionToTimeWithResult.ID))
```

Comment:
I recommend we use static import for `assertThat`

Reply:
Done.